### PR TITLE
ROI tab revamped

### PR DESCRIPTION
**Pull request by maharshi-gor**
### ROIs tab revamped.

**Goals**

1. This PR makes the ROI tab follow the required APIs by Horizon Tab according to the new architecture.
2. This also introduces a opacity toggle for ROIs.
Screenshot from 2024-03-21 11-31-14

### CODE REVIEW

File: dipy/viz/horizon/tab/roi.py [hunk 1]
```diff
@@ -1,52 +1,106 @@
...
 
 class ROIsTab(HorizonTab):
     def __init__(self, contour_actors):
-        self.__actors = contour_actors
-        self.__name = 'ROIs'
+        """Initialize interaction tab for ROIs visualization.
+
+        Parameters
+        ----------
+        contour_actors : list
```
Comment (skoudoro) [inline dipy/viz/horizon/tab/roi.py:10]: list of ? string ? actors? 
Can you add a description below also

File: dipy/viz/horizon/tab/roi.py [hunk 2]
```diff
@@ -1,52 +1,106 @@
...
 
-    def build(self, tab_id, tab_ui):
-        self.__tab_id = tab_id
-        self.__tab_ui = tab_ui
+    def build(self, tab_id, _tab_ui):
+        """Position the elements in the tab.
+
+        Parameters
+        ----------
+        tab_id : int
```
Comment (skoudoro) [inline dipy/viz/horizon/tab/roi.py:67]: Can you add a description

File: dipy/viz/horizon/tab/roi.py [hunk 3]
```diff
@@ -1,52 +1,106 @@
...
+
+        Returns
+        -------
+        str
+        """
+        return self._name
+
+    @property
+    def actors(self):
+        """Visualization actors controlled by tab.
```
Comment (skoudoro) [inline dipy/viz/horizon/tab/roi.py:91]: Visualization ? Strange description

**Review by skoudoro — COMMENTED** [review]
Hi @maharshi-gor,

Thank you for this. Can you make Codecov happy also: https://github.com/dipy/dipy/pull/3144#issuecomment-2012996134

Can you squash your commit and start to use commit codes : https://docs.dipy.org/stable/devel/commit_codes.html

See below for some additional comments

**Comment by skoudoro** [comment]
Hi @maharshi-gor, 

To add to my review above, Can you also fix the conflict ?

thanks !

File: dipy/viz/horizon/tab/roi.py (hunk 1 above)
Comment (maharshi-gor) [reply dipy/viz/horizon/tab/roi.py:10]: Updated!

File: dipy/viz/horizon/tab/roi.py (hunk 3 above)
Comment (maharshi-gor) [reply dipy/viz/horizon/tab/roi.py:91]: Yes it does not contain the UI actors but the actors you see in the visualization of the specific rois

**Review by skoudoro — APPROVED** [review]
Thank you for this @maharshi-gor,  Looks good, merging

**Pull request merged**